pkg/keyspace/tso_keyspace_group.go

```diff
@@ -712,6 +702,11 @@ func (m *GroupManager) FinishSplitKeyspaceByID(splitTargetID uint32) error {
 	var splitTargetKg, splitSourceKg *endpoint.KeyspaceGroup
 	m.Lock()
 	defer m.Unlock()
+
+	failpoint.Inject("pauseTxn", func() {
+		time.Sleep(2 * time.Second)
+	})
+
```

Comment:
If the test failed without the 2-second pause, will the production system fail as well?
* If yes, then only injecting the failpoint doesn't fix the production.
* If no, then what's the difference between the integration test and the production system that caused such diversion?

Reply:
it is not related about production system, it is just about this test case. This test case will check whether the primarys before splitting finish, but not well control the race condition